api/src/main/java/org/apache/iceberg/expressions/UnboundPredicate.java

```diff
@@ -284,6 +287,14 @@ public String toString() {
         return term() + " startsWith \"" + literal() + "\"";
       case NOT_STARTS_WITH:
         return term() + " notStartsWith \"" + literal() + "\"";
+      case ST_INTERSECTS:
+      case ST_DISJOINT:
+        {
+          ByteBuffer serializedBoundingBox = (ByteBuffer) literal().value();
```

Comment:
Looking at the type cast and fromByteBuffer, I am wondering if the `Literal.of(BoundingBox)` should return `BoundingBox` type (instead of `ByteBuffer`.

`BoundingBoxLiteral` should probably implement `toString` method to simplify the code here.

Reply:
Implemented toString for `BoundingBoxLiteral`.